sql/core/src/main/scala/org/apache/spark/sql/streaming/StreamingQuery.scala

```diff
@@ -142,10 +143,19 @@ trait StreamingQuery {
   def processAllAvailable(): Unit
 
   /**
-   * Stops the execution of this query if it is running. This method blocks until the threads
-   * performing execution has stopped.
+   * Stops the execution of this query if it is running. This awaits the termination of the query
```

Comment:
super nit: `This waits until the termination ... or until a timeout ....`
The parallel structure is weird in the current form.